app/assets/javascripts/ama_layout/confirm_modal.js

```diff
@@ -45,12 +51,33 @@ $.rails.showConfirmationDialog = function(link){
     <a href="#" data-reveal-confirm="my-data-element">OK</a>
     ...
   </div>
+
+  Multiple Links/Single Modal:
+
+  Links
+  <a href="somewhere" data-confirm="my-data-element" data-reveal-confirm-key="element-1">Delete</a>
+  <a href="somewhere-else" data-confirm="my-data-element" data-reveal-confirm-key="element-2">Delete</a>
+
+  Modal
+  <div class="reveal" data-reveal data-my-data-element>
+    ...
+    <a href="#" data-reveal-confirm="my-data-element">OK</a>
+    ...
+  </div>
 */
-$(document).on('click', '*[data-reveal-confirm]', function(e){
+$(document).on('click', '*[data-reveal-confirm]', function(e) {
   e.preventDefault();
-  var el = $(e.currentTarget).data('reveal-confirm');
-  var link = $('[data-confirm="'+ el + '"]');
+  var link;
+  var target = e.currentTarget;
+  var el = $(target).data('reveal-confirm');
   var modal = $('[data-' + el + ']');
+  var key = $(modal).data('reveal-confirm-key');
+  if (key && key.length) {
+    link = $('[data-reveal-confirm-key=' + key + ']');
+  } else {
+    link = $('[data-confirm="'+ el + '"]');
+  }
```

Comment:
What do you think about:
`var link = $('[data-confirm="'+ el + '"]');`
`if (key && key.length) {`
` link = $('[data-reveal-confirm-key=' + key + ']');`
 `}`

Reply:
Yes - this is better.